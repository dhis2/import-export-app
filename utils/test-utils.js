import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
