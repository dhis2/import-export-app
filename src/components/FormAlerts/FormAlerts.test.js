import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { FormAlerts } from '../index.js'

const alerts = [
    { id: '0', message: 'Alert 0 - info', info: true },
    { id: '1', message: 'Alert 1 - warning', warning: true },
    { id: '2', message: 'Alert 2 - critical', critical: true },
    { id: '3', message: 'Alert 3 - success', success: true },
]

const props = {
    dataTest: 'form-alerts',
    alerts: [],
}

it('matches snapshot when no alerts', () => {
    const { asFragment } = render(<FormAlerts {...props} />)
    expect(asFragment()).toMatchSnapshot()
})

it('matches snapshot when multiple alerts', () => {
    const { asFragment } = render(<FormAlerts {...props} alerts={alerts} />)
    expect(asFragment()).toMatchSnapshot()
})
