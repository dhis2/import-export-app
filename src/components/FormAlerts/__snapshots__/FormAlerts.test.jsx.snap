// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when multiple alerts 1`] = `<DocumentFragment />`;

exports[`matches snapshot when no alerts 1`] = `<DocumentFragment />`;
